node-native/src/types.rs

```diff
@@ -0,0 +1,421 @@
+use libp2p::swarm::ConnectionCounters as Libp2pConnectionCounters;
+use libp2p::swarm::NetworkInfo as Libp2pNetworkInfo;
+use libp2p::PeerId as Libp2pPeerId;
+use lumina_node::block_ranges::BlockRange as LuminaBlockRange;
+use lumina_node::events::{NodeEvent as LuminaNodeEvent, NodeEventInfo as LuminaNodeEventInfo};
+use lumina_node::network;
+use lumina_node::node::PeerTrackerInfo as LuminaPeerTrackerInfo;
+use lumina_node::node::SyncingInfo as LuminaSyncingInfo;
+use std::str::FromStr;
+use std::time::SystemTime;
+use uniffi::{Enum, Record};
+
+/// Supported Celestia networks.
+#[derive(Debug, Default, Clone, Enum)]
+pub enum Network {
+    /// Celestia mainnet.
+    #[default]
+    Mainnet,
+    /// Arabica testnet.
+    Arabica,
+    /// Mocha testnet.
+    Mocha,
+    /// Custom network.
+    Custom { id: String },
+}
+
+// From implementation for converting between Lumina and Uniffi types
+impl From<&Network> for network::Network {
+    fn from(network: &Network) -> Self {
+        match network {
+            Network::Mainnet => network::Network::Mainnet,
+            Network::Arabica => network::Network::Arabica,
+            Network::Mocha => network::Network::Mocha,
+            Network::Custom { id } => {
+                network::Network::Custom(network::NetworkId::new(id).expect("invalid network id"))
+            }
+        }
+    }
+}
+
+/// Statistics of the connected peers
+#[derive(Debug, Clone, Default, Record)]
+pub struct PeerTrackerInfo {
```

Comment:
we were thinking to add `uniffi` feature in both `celestia-types` and `lumina-node` crates and try to wrap as many stuff as possible directly instead of reintroducing this here. Is there any drawback with such approach? We don't have much experience with uniffi but [here](https://mozilla.github.io/uniffi-rs/latest/tutorial/Rust_scaffolding.html) it mentions that it should be possible with `uniffi_reexport_scaffolding`